fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -165,19 +187,28 @@ public Map<String, HiveColumnStats> getTableLevelColumnStats(List<String> column
 
     @Override
     public void refreshTableCache(String dbName, String tableName) throws DdlException {
-        GlobalStateMgr.getCurrentState().getHiveRepository().refreshTableCache(hmsTableInfo);
+        HiveRepository hiveRepository = GlobalStateMgr.getCurrentState().getHiveRepository();
+        try {
+            if (HiveMetaStoreTableUtils.isInternalCatalog(resourceName)) {
+                hiveRepository.refreshTableCache(hmsTableInfo);
+            } else {
+                hiveRepository.refreshConnectorTable(resourceName, dbName, tableName);
+            }
```

Comment:
As HiveTable, internal catalog needs to refreshTableCache, and connector table needs to refreshConnectorTable.